data/abilities.ts

```diff
@@ -184,8 +184,12 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 			this.add('-ability', pokemon, 'Unnerve');
 			this.effectState.unnerved = true;
 		},
-		onEnd() {
+		onEnd(pokemon) {
 			this.effectState.unnerved = false;
+			if (this.effectState.unnerved) return;
```

Comment:
the unnerve changes were from another pr

Reply:
edited the comment, also this pr refactors healing berries on top of confusion interaction

Reply:
if two unnerve pokemon are on the field, don't eat the berry because when a pokemon hits itself in confusion it eats the berry either when it takes damage or when all opposing unnerve pokemon leave the field

Reply:
oh right this broke when I changed it from onUpdate. I can do that. Is each instance of unnerve on the field going to change the same effect state or is there one for each pokemon

Reply:
the confusion self hit flag is set to false whenever unnerve leaves the field, I did not change any existing unnerve implementations